project/genprod.scala

```diff
@@ -15,7 +15,7 @@
  * where 0 <= N <= MaxArity. Usage: sbt generateSources
  */
 object genprod {
-  final val MaxArity = 22
+  final val MaxArity = 23
```

Comment:
IIRC, when more than 22, it converts to array.

Reply:
See you'd reckon so, but checked this out and this is the current behaviour:

```
scala> val a = (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23)
<console>:1: error: too many elements for tuple: 23, allowed: 22
val a = (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23)
        ^
```